util/json_helper.go

```diff
@@ -0,0 +1,69 @@
+package util
+
+import (
+	"encoding/json"
+	"fmt"
+	"reflect"
+	"strings"
+)
+
+// JSONMarshal marshals an arbitrary map to JSON; this only exists because
+// JSON.Marshal insists on having maps that have string (and not interface{})
+// keys
+func JSONMarshal(input interface{}) ([]byte, error) {
+	converted, err := jsonMarshalHelper(input)
+	if err != nil {
+		return nil, err
+	}
+	return json.Marshal(converted)
+}
+
+type jsonMarshalError struct {
+	msg  string
+	keys []string
+}
+
+func (e jsonMarshalError) Error() string {
+	return fmt.Sprintf("Error marshalling JSON: Failed to convert keys in path %s: %s",
+		strings.Join(e.keys, "."), e.msg)
+}
+
+// jsonMarshalHelper converts a map from having interface{} keys to string keys
+func jsonMarshalHelper(input interface{}) (interface{}, *jsonMarshalError) {
+	value := reflect.ValueOf(input)
+	switch value.Kind() {
+	case reflect.Map:
+		if value.Kind() != reflect.Map {
```

Comment:
Is this even possible given that we are in the `reflect.Map` branch of a switch testing `value.Kind()` ?